include/xtensor/xnpy.hpp

```diff
@@ -42,17 +43,15 @@ namespace xt
     namespace detail
     {
 
-    /* Compile-time test for byte order.
-       If your compiler does not define these per default, you may want to define
-       one of these constants manually.
-       Defaults to little endian order. */
-#if defined(__BYTE_ORDER) && __BYTE_ORDER == __BIG_ENDIAN || defined(__BIG_ENDIAN__) || \
-    defined(__ARMEB__) || defined(__THUMBEB__) || defined(__AARCH64EB__) ||             \
-    defined(_MIBSEB) || defined(__MIBSEB) || defined(__MIBSEB__)
-        const bool big_endian = true;
-#else
-        const bool big_endian = false;
-#endif
+        /* Test for endianess. Compiler can optimize that to a single constant. */
+        static inline bool is_big_endian()
+        {
+            uint32_t utmp = 0x01020304;
+            char btmp[sizeof(utmp)];
+            std::memcpy(&btmp[0], &utmp, sizeof(utmp));
+            const bool big_endian = btmp[0] == 0x01;
+            return big_endian;
+        }
```

Comment:
Should we have `is_big_endian()` in a common place? This way we could also use it e.g. in xtensor-io instead of [this](https://github.com/xtensor-stack/xtensor-io/blob/c2235543a595134ac953153fc69a8e3d66d60cc8/include/xtensor-io/xtensor-io.hpp#L29-L37).